src/pages/layout.js

```diff
@@ -10,76 +10,189 @@ import i18n from '@dhis2/d2-i18n'
 
 import { D2Shim } from '../utils/D2Shim'
 import layoutStyles from '../styles/Layout.module.css'
+import api from '../utils/api'
+import {
+    DATASET_SETTINGS,
+    GENERAL_SETTINGS,
+    NAMESPACE,
+    PROGRAM_SETTINGS,
+} from '../constants/data-store'
+import { androidSettingsDefault } from '../constants/android-settings'
+import { SpecificSettingsDefault } from '../constants/program-settings'
+import { DataSetSettingsDefault } from '../constants/data-set-settings'
+import DialogFirstLaunch from '../components/dialog/dialog-first-launch'
 
 const styles = {
     twoPanelMain: {
         marginTop: '0rem',
     },
 }
 
-const Layout = props => {
-    const currentSection = useRef()
-    const sidebarRef = useRef()
-
-    const routeSections = menuSection
-    routeSections.push({
-        key: 'home',
-        path: '/',
-        component: <AndroidSettingsContainer />,
-    })
-
-    const changeSectionHandler = key => {
-        currentSection.current = key
-        if (key !== 'search' && sidebarRef.current) {
-            sidebarRef.current.clearSearchBox()
-        }
+const DEFAULT_PROGRAM = 'DEFAULT_PROGRAM'
+const DEFAULT_DATASET = 'DEFAULT_DATASET'
+
+const populateObject = (type, settingsList) => {
```

Comment:
How about instead of passing `settingsList`, you just use the correct object (either DataSetSettingsDefault or SpecificSettingsDefault) based on the `type`?

Reply:
Seems like a great idea, I just solved it